User feedback tool

Mage-Huesmann opened the pull request:
Global user feedback utility tool with Admin dashboard viewing

Commit:
New branch for Global user feedback and admin viewing

Commit:
Working global feedback tool, viewable by admins

Commit:
Feedback done?

Commit:
Dashboard upgrades

Commit:
File attachment in feedback tool

Commit:
requirements lagged, adding now

moom359 commented:
1) We either need to standardize on the services in docker compose either having or not having the "_service" suffix. I am personally against _service. I suggest it be removed for feedback service.

2) The API Gateway PathPrefixes for feedback should not have the v1. Standardize to /api/feedback and do not define a new router just for attachments (expand the rules for feedback service to include something like /api/feedback/attachments or /api/fb/attachments if you really want to differentiate them). This entails changes to dynamic-conf and some frontend files. You also need to add auth-middleware to your middlewares section of dynamic-conf.

3) Port 8010 is not available for feedback service. It is already claimed by Auth.

4) Feedback service app.py; remove the prefix from line 19 (app.include_routers(...)). The standard for this project is to avoid defining prefixes and instead verbosely define paths so that the code is more readable and debugging is easier.

5) FeedbackErrorBoundry.js line 217; the error message shouldn't say "check your internet connection" since we are targeting air gapped environments. Probably just change to something like "check the feedback container".

6) FeedbackErrorBoundry.js line 222 & 235; Either add a call to the logger to actually log the error or at least add a TODO so we can catch that we need to add it in.

7) Nit-noid thing, but take the emoji that cursor added to your console.log in feedbackfileupload.js:308. Similarly, probably get rid of them in mage-dev.sh; we will probably run into systems that don't support the emoji characters and it could break the script.

Mage-Huesmann commented:
New Pull request inbound with updates as follows

1. No code change, discussion needed

2. "v1" removed from feedback service, auth middleware added

3. Non-issue

4. Prefix removed

5. Fixed error message

6. TODOs added in code 

7. No code change, fixed in another branch
